### PR TITLE
Add workflow file for TPU CI

### PR DESCRIPTION
**Pull request by mbzomowski**
Adding a workflow file for TPU Continuous Integration. This workflow file will run integration tests on a TPU on demand (workflow_dispatch) for PR's, and on a regular schedule.

### CODE REVIEW

File: .github/workflows/tpu_ci.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: TPU Integration Test
+run-name: TPU Testing
+on:
+  workflow_dispatch:
+  schedule:
```
Comment (will-cromar) [inline .github/workflows/tpu_ci.yml:5]: Can you run this on push to `master` instead of on a schedule?

**Review by will-cromar — COMMENTED** [review]
Nice! I want to see if it's possible to add TPU support to our existing build workflow, then kick off this workflow with that wheel. This is a great start!

File: .github/workflows/tpu_ci.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          TPUVM_MODE: 1
+        run: |
+          cd pytorch/xla
+          python3 setup.py install --user
+      - name: Run Tests
+        env:
+          PJRT_DEVICE: TPU
+        run: |
+          cd pytorch/xla
+          python3 -u test/test_operations.py -v
```
Comment (will-cromar) [inline .github/workflows/tpu_ci.yml:33]: Can you add the rest of the tests from [here](https://github.com/pytorch/xla/blob/d6cabc5a45944365c0f3e6969ac4083eb425a5a3/test/tpu/xla_test_job.yaml#L47-L59), or move those tests into a script and invoke it from both places?

File: .github/workflows/tpu_ci.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+name: TPU Integration Test
+run-name: TPU Testing
+on:
+  workflow_dispatch:
```
[the remark this answers is not in the document]
Comment (mbzomowski) [reply .github/workflows/tpu_ci.yml:4]: https://github.com/mbzomowski/xla/actions/runs/7252158941

File: .github/workflows/tpu_ci.yml (hunk 2 above)
Comment (mbzomowski) [reply .github/workflows/tpu_ci.yml:34]: Added

File: .github/workflows/tpu_ci.yml (hunk 1 above)
Comment (mbzomowski) [reply .github/workflows/tpu_ci.yml]: Done

File: .github/workflows/tpu_ci.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (mbzomowski) [reply .github/workflows/tpu_ci.yml:4]: Test run with the additional tests now: https://github.com/mbzomowski/xla/actions/runs/7254723255/job/19763990461

**Comment by mbzomowski** [comment]
Tests passing on forked repo: https://github.com/mbzomowski/xla/actions/runs/7401813748/job/20138389743#logs

**Review by will-cromar — APPROVED** [review]
Nice! Great work!

**Pull request merged**